doc/decap/subnet_decap_HLD.md

```diff
@@ -0,0 +1,194 @@
+| source IP mask  | Netscan privately-owned subnet mask |                                                                                                                          |
+
+### 6.2 DB Schema
+
+#### 6.2.1 CONFIG_DB
+
+```
+### SUBNET_DECAP
+    ; Stores subnet based decapsulation configurations
+    key                 = SUBNET_DECAP|subnet_type                      ; SUBNET_DECAP|vlan
+    status              = "enable"/"disable"                            ; status of subnet based decapsulation
+    src_ip              = source IP prefix                              ; source IP prefix used for decap terms of IPv4 tunnel
+    src_ip_v6           = source IP prefix                              ; source IPv6 prefix used for decap terms of IPv6 tunnel
+```
+
+#### 6.2.2 APPL_DB
+
+```
+### TUNNEL_DECAP_TABLE
+    ; Stores a list of decap tunnels
+    key                 = TUNNEL_DECAP_TABLE:tunnel_name                ; tunnel name as key
+    tunnel_type         = "IPINIP"                                      ; tunnel type
+    dscp_mode           = "uniform"/"pipe"
+    ecn_mode            = "copy_from_outer"/"standard"
+    ttl_mode            = "uniform"/"pipe"
+    encap_ecn_mode      = "standard"
+
+### TUNNEL_DECAP_TERM_TABLE
+    ; Stores a list of decap terms.
+    key                 = TUNNEL_DECAP_TERM_TABLE:tunnel_name:dst_ip    ; tunnel name:dst IP prefix as key
+    term_type           = "P2P"/"P2MP"/"MP2MP"                          ; tunnel decap term type
+    src_ip              = source IP prefix
+```
+
+#### 6.2.3 STATE_DB
+
+```
+### TUNNEL_DECAP_TABLE
+    ; Stores a list of created decap tunnels
+    key                 = TUNNEL_DECAP_TABLE:tunnel_name                ; tunnel name as key
+    tunnel_type         = "IPINIP"                                      ; tunnel type
+    dscp_mode           = "uniform"/"pipe"
+    ecn_mode            = "copy_from_outer"/"standard"
+    ttl_mode            = "uniform"/"pipe"
+    encap_ecn_mode      = "standard"
+
+### TUNNEL_DECAP_TERM_TABLE
+    ; Stores a list of created decap terms.
+    key                 = TUNNEL_DECAP_TERM_TABLE:tunnel_name:dst_ip    ; tunnel name:dst IP prefix as key
+    term_type           = "P2P"/"P2MP"/"MP2MP"                          ; tunnel decap term type
+    src_ip              = source IP prefix
+```
+
+### 6.3 Orchestration Agent
+
+The following orchestration agents shall be modified:
+TunnelDecapOrch:
+
+- TunnelDecapOrch shall subscribe to TUNNEL_DECAP_TABLE and create/remove the decap tunnels.
+
+- TunnelDecapOrch shall subscribe to TUNNEL_DECAP_TERM_TABLE and create/remove the decap term entries.
+
+- TunnelDecapOrch shall subscribe to SUBNET_DECAP and handle the tunnel decapsulation termination entry source IP changes.
+
+### 6.4 VLAN Subnet Decap
+
+#### 6.4.1 VLAN Subnet Decap Rule Generation
```

Comment:
since swssconfig is oneshot script, how to handle the change of vlan interface address after swssconfig playing?

Reply:
That is not supported/covered by this HLD